homeassistant/components/fully_kiosk/camera.py

```diff
@@ -0,0 +1,105 @@
+"""Support for Fully Kiosk Browser camera."""
+
+from __future__ import annotations
+
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+from typing import Any
+
+from fullykiosk import FullyKiosk
+
+from homeassistant.components.camera import (
+    Camera,
+    CameraEntityDescription,
+    CameraEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import FullyKioskDataUpdateCoordinator
+from .entity import FullyKioskEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class FullyCameraEntityDescription(CameraEntityDescription):
+    """Fully Kiosk Browser camera entity description."""
+
+    image_action: Callable[[FullyKiosk], Coroutine[Any, Any, bytes]]
+    on_action: Callable[[FullyKiosk], Any] | None
+    off_action: Callable[[FullyKiosk], Any] | None
+    is_on_fn: Callable[[dict[str, Any]], Any]
+
+
+CAMERAS: tuple[FullyCameraEntityDescription, ...] = (
+    FullyCameraEntityDescription(
+        key="camera",
+        translation_key="camera",
+        image_action=lambda fully: fully.getCamshot(),
+        on_action=lambda fully: fully.enableMotionDetection(),
+        off_action=lambda fully: fully.disableMotionDetection(),
+        is_on_fn=lambda data: data["settings"].get("motionDetection"),
+    ),
+    FullyCameraEntityDescription(
+        key="screenshot",
+        translation_key="screenshot",
+        image_action=lambda fully: fully.getScreenshot(),
+        on_action=None,
+        off_action=None,
+        is_on_fn=lambda data: True,
+    ),
```

Comment:
A screenshot is not a camera, for this one, an image entity should be used instead.

Reply:
Removed for now and I might add it in a separate PR.